examples/singlefile/app.py

```diff
@@ -0,0 +1,150 @@
+"""A single-file demo of pyramid_openapi3.
+
+Usage:
+* git clone https://github.com/niteoweb/pyramid_openapi3.git
+* cd pyramid_openapi3/examples/singlefile
+* virtualenv -p python3.7 .
+* source bin/activate
+* pip install pyramid_openapi3
+* python app.py
+"""
+
+from pyramid.config import Configurator
+from pyramid.httpexceptions import HTTPForbidden
+from pyramid.view import view_config
+from wsgiref.simple_server import make_server
+
+import tempfile
+import unittest
+
+# This is usually in a separate openapi.yaml file, but for the sake of the
+# example we want everything in a single file. Other examples have it nicely
+# separated.
+OPENAPI_DOCUMENT = b"""
+    openapi: "3.0.0"
+    info:
+      version: "1.0.0"
+      title: Hello API
+    paths:
+      /hello:
+        get:
+          parameters:
+            - name: name
+              in: query
+              required: true
+              schema:
+                type: string
+                minLength: 3
+          responses:
+            200:
+              description: Say hello
+"""
+
+
+@view_config(route_name="hello", renderer="json", request_method="GET", openapi=True)
+def hello(request):
+    """Say hello."""
+    if request.openapi_validated.parameters["query"]["name"] == "admin":
+        raise HTTPForbidden()
+    return {"hello": request.openapi_validated.parameters["query"]["name"]}
+
+
+def app(spec):
+    """Prepare a Pyramid app."""
+    with Configurator() as config:
+        config.include("pyramid_openapi3")
+        config.pyramid_openapi3_spec(spec)
+        config.pyramid_openapi3_add_explorer()
+        config.add_route("hello", "/hello")
+        config.scan(".")
+        return config.make_wsgi_app()
+
+
+if __name__ == "__main__":
+    """If app.py is called directly, start up the app."""
+    with tempfile.NamedTemporaryFile() as document:
+        document.write(OPENAPI_DOCUMENT)
```

Comment:
Swagger client examples require the path to JSON spec https://github.com/swagger-api/swagger-js#import-in-browser. I think is better to expose JSON and yaml as routes, so that it can actually be used.

Reply:
pyramid_openapi3 takes care of publishing the YAML file:

<img width="420" alt="Screenshot 2019-04-14 at 20 08 27" src="https://user-images.githubusercontent.com/311580/56097817-2e64d480-5ef1-11e9-9bbf-2520f10a4458.png">

I double checked, and the try-me examples seem to work:

<img width="835" alt="Screenshot 2019-04-14 at 20 08 48" src="https://user-images.githubusercontent.com/311580/56097823-3de41d80-5ef1-11e9-97bf-d8f742f2dd33.png">

Or am I misunderstanding what you are trying to say?

Reply:
I was thinking if you use API outside of API explorer.

Reply:
So, to convert YAML into JSON and publish both?

Reply:
Yep, so add another handler for JSON request.
https://github.com/niteoweb/pyramid_openapi3/blob/master/pyramid_openapi3/__init__.py#L133

See example in https://github.com/swagger-api/swagger-js#in-browser

Reply:
That's completely out of the scope of this PR.

https://github.com/niteoweb/pyramid_openapi3/issues/10